src/components/file/File.js

```diff
@@ -99,30 +100,63 @@ export class File extends React.Component {
     }
   }
 
+  renderCopyButton = ({ url, t }) => {
+    if (this.props.isDownload) {
+      return null
+    }
+
+    const copyBtnClass = classnames({
+      'o-50 no-pointer-events': this.state.copied,
+      'o-80 glow pointer': !this.state.copied
+    }, ['pa2 w3 flex items-center justify-center br-pill bg-aqua f7 fw5'])
+
+    return (
+      <CopyToClipboard text={url} onCopy={this.handleOnCopyClick}>
+        <div className={copyBtnClass}>
+          {this.state.copied ? t('copyLink.copied') : t('copyLink.copy')}
+        </div>
+      </CopyToClipboard>
+    )
+  }
+
   render () {
     const { name, type, shareCID, error, t } = this.props
     const size = filesize(this.props.size || 0, { round: 0, spacer: '' })
 
     const fileNameClass = classnames({ charcoal: !error, gray: error }, ['FileLinkName ph2 f6 b truncate'])
     const fileSizeClass = classnames({ 'charcoal-muted': !error, gray: error }, ['f6'])
 
+    const url = `${ENDPOINTS.gateway}/${shareCID}?filename=${encodeURI(name)}`
```

Comment:
When `?filename` is used we need to replace `shareCID` (CID of wrapper  directory) with the CID of the file.
I think its `cid` inside of `this.props`?

perhaps 

```diff
-      const { name, type, shareCID, error, t } = this.props 
+     const { name, type, cid, error, t } = this.props 
```

And then `cid.toString('base32')` (to stabilize CID so it looks the same on path and subdomain)

Reply:
correct, it's cid

Reply:
This makes sense, but I'm afraid I can't find anywhere to `cid.toString('base32')` without getting "undefined" errors.

Reply:
Seems that `cid` is not always present when render is executed –  this becames a problem if we want to calle `toV1` on it (to ensure its always something we can represent as base32), so I just added additional handling – should work fine now.